### PR TITLE
Parallelize encryption to improve performance

### PR DESCRIPTION
**Pull request by burke**
@Shopify/locutus 

Turns out encryption is pretty slow.

yak

Time to decrypt test file fetched from  https://edg.epa.gov/data.json and encrypted with a random key:

On my macbook:

* Go 1.4, GOMAXPROCS=1, master branch: 7.8s
* Go 1.5, GOMAXPROCS=1, master branch: 5.8s
* Go 1.5, GOMAXPROCS=1, this branch: 6.9s
* Go 1.5, GOMAXPROCS=8, this branch: 1.9s

On production infrastructure (first is Go 1.4, 1 proc, master; second is Go 1.5, 32 procs, optimize branch):

```
burke@borg3.ash:~ $ ./test_deployed
7.371902681s
burke@borg3.ash:~ $ ./test_go_1.5_optimize_branch
794.795795ms
```

### CODE REVIEW

File: cmd/ejson/main.go [hunk 1]
```diff
@@ -16,6 +17,8 @@ func execManpage(sec, page string) {
 }
 
 func main() {
+	// Encryption is expensive. We'd rather burn cycles on many cores than wait.
+	runtime.GOMAXPROCS(runtime.NumCPU())
```
Comment (csfrancis) [inline cmd/ejson/main.go]: It might be nice to make the default use all cores and allow the command line to override with another value. But whatever.

Comment (burke) [inline cmd/ejson/main.go]: Meh, I want ejson to have as few options as possible. I want to pick one way and stick to it.

**Comment by burke** [comment]
Damn, turns out the profile result I was originally looking at that motivated this was from when we were still invoking ejson as `bundle exec ejson`. Our ejson files aren't actually big enough that this really pays off; most of the cost was from loading bundler and ruby. This'll only save us like 50-100ms.

**Comment by csfrancis** [comment]
Oh that's too bad. It's still pretty cool though.

:+1:

File: cmd/ejson/main.go (hunk 1 above)
Comment (csfrancis) [inline cmd/ejson/main.go]: :+1:

**Pull request merged**